support-frontend/conf/routes

```diff
@@ -45,23 +45,23 @@ GET /event/*eventId                                 controllers.Application.redi
 
 GET  /monthly-contributions                         controllers.Application.permanentRedirect(location="/contribute")
 
-# Deprecated endpoints that we now redirect to the contribute page. Once we have established that no traffic
-# is coming this way, we should remove them
-GET  /contribute/recurring                          controllers.Application.permanentRedirect(location="/contribute")
-GET  /contribute/recurring-guest                    controllers.Application.permanentRedirect(location="/contribute")
-GET  /contribute/recurring/thankyou                 controllers.Application.permanentRedirect(location="/contribute")
-GET  /contribute/recurring/pending                  controllers.Application.permanentRedirect(location="/contribute")
-GET  /contribute/recurring/existing                 controllers.Application.permanentRedirect(location="/contribute")
-GET  /contribute/one-off                            controllers.Application.permanentRedirect(location="/contribute")
-GET  /contribute/one-off/thankyou                   controllers.Application.permanentRedirect(location="/contribute")
-GET  /contribute/one-off/autofill                   controllers.Application.permanentRedirect(location="/contribute")
-
 
 # ----- Contributions ----- #
 
-GET  /contribute                                        controllers.Application.contributeGeoRedirect()
-GET  /$country<(uk|us|au|eu|int|nz|ca)>/contribute      controllers.Application.contributionsLanding(country: String)
-GET  /$country<(uk|us|au|eu|int|nz|ca)>/thankyou        controllers.Application.contributionsLanding(country: String)
+GET  /contribute                                    controllers.Application.contributeGeoRedirect(campaignCode = "")
+GET  /$country<(uk|us|au|eu|int|nz|ca)>/contribute  controllers.Application.contributionsLanding(country: String)
+
+# ----- Toxic America Campaign ---- #
+GET  /contribute/toxicamerica                       controllers.Application.contributeGeoRedirect(campaignCode = "toxicamerica")
```

Comment:
looking at this, I wonder if it would be simpler with just `GET    /contribute/:campaignCode` and a pattern match in the controller action, falling back to the georedirect. Ditto for the country-specific one. You can then get rid of the wildcard redirect routes, and it's clearer that those wildcard ones are really intended to cover deprecated campaigns

Reply:
We had that setup before but the controller with the if statement got pretty messy because it needed to return a future (from what I remember). I think this approach is pretty declarative and lets us change out the endpoint when there's a new campaign in a more obvious place.

Reply:
@joelochlann We did have that at one point. It's basically a toss up between logic in the controller or logic in the routes file. I preferred this in the end but there's not much between them. I don't think either is necessarily simpler, I think both solutions basically implement the same logic, it's about which is clearer. 

`it's clearer that those wildcard ones are really intended to cover deprecated campaigns`

This would be an argument for what you are suggesting. The way we implemented it before was by pointing `GET /contribute/:campaignCode` to the contributions Landing controller, and made that take a campaign code parameter. This felt messy as there was suddenly quite  a lot of logic in that controller, as it had to check to see if there was a campaign code and if there was, whether it matched the whitelisted campaign code. But maybe adding a seperate controller for  `GET /contribute/:campaignCode` would solve this problem. When Joshua comes in I think we'll give `GET /contribute/:campaignCode` another whack

Reply:
Just saw that Joshua isn't going to be in today so I'll have a go myself

Reply:
What's the advantage of having the server whitelist campaign codes?
If the path component after `/contribute/` is always used for campaign codes then the client can check it, and ignore it if it's not a genuine campaign. Or is that crazy?

Reply:
Ok @tomrf1 and @joelochlann I've simplified things

Reply:
nice!